### ISSUE TITLE
[Firefox] Tab selector list error?

### ISSUE THREAD

**Issue by alejandrolemus** [gorhill/uBlock#212]
Firefox 38.0.1, uBlock0 9.7.1-dev1:
Created a new tab and accordingly a new list item <code>about:newtab</code> appeared in the tab selector. The tab was closed later without actually requesting a site, but the the item remained in the tabs list. A browser restart didn't fix it.

Edit: now I also see an <code>about:customizing</code> item in the list. I'm guessing it was the Firefox about dialog I opened to check browser version while posting? Confusing.

Edit 2: I also noticed a self-repair.mozilla.org url (see image). Shouldn't that be a behind-the scene request? It's not a tab, so it shouldn't be on the tabs selector. 

![ublock](https://cloud.githubusercontent.com/assets/9505902/7671727/8a154a1c-fca8-11e4-8ab0-efa74a32f064.png)

**Comment by gorhill** [comment]
Firefox/uBlock quirk.

This quirk was causing [issue #189](https://github.com/gorhill/uMatrix/issues/189) in uMatrix. They are harmless. I tried to port the fix from uMatrix but that did not work, and this would risk breaking Fennec support, for which I have no way to test.

**Comment by alejandrolemus** [comment]
They are all gone now, don't know what triggered the refresh. But happens again as soon as a new tab is created. Hopefully you can come up with a fix at some point. Thanks!

**Comment by gorhill** [comment]
> They are all gone now, don't know what triggered the refresh

There is a [garbage collector](https://github.com/gorhill/uBlock/blob/facef0dc054f44f56fef38c5acf93d1748cfe7db/src/js/tab.js#L634) in uBlock to get rid of stale tab entries in its internal data structure, so it's probably what kicked in (currently set at every 7 minutes, checks at most 11 tabs each time -- some users have 100s of tabs opened...)

The other uBlock also has the issue, however you do not see them in the tab selector because I changed here the way the tab entries are reported to the logger, to avoid the issue of potential overhead which was discussed in <https://github.com/chrisaljoudi/uBlock/issues/484#issuecomment-69577460>.

Long story short, uBlock Origin got rid of the overhead worries, but then will report every tabs in its internal data structure, including those which may have been removed by the browser (without uBlock being notified about it).

An ideal fix would be for FF to notify uBlock that the temporary tabs have been destroyed. I will dig more into this eventually.

**Comment by alejandrolemus** [comment]
Just to confirm that fix [6470216](https://github.com/gorhill/uBlock/commit/6470216530548cd3dfecdce7d9aa94816d10d863) worked great.
